docker/dependencies/Dockerfile

```diff
@@ -0,0 +1,245 @@
+        # CI dependencies
+        git ssh tar gzip ca-certificates gnupg \
+        # Pythons
+        python-pip python3-pip python3-setuptools python-dev \
+        # SWIG dependencies
+        libpcre3-dev autoconf bison  \
+        # other
+        wget curl file gdb ccache \
+        gcovr cppcheck doxygen graphviz graphviz-dev unzip zip; \
+    apt-get -y clean
+
+# install cmake 3.10.2
+RUN set -e; \
+    git clone https://gitlab.kitware.com/cmake/cmake.git /tmp/cmake; \
+    (cd /tmp/cmake ; git checkout c1e087a9d3af74299d7681c9f9de59e5977a1539); \
+    (cd /tmp/cmake ; /tmp/cmake/bootstrap --system-curl --parallel=${PARALLELISM} --enable-ccache); \
+    make -j${PARALLELISM} -C /tmp/cmake; \
+    make -C /tmp/cmake install; \
+    ldconfig; \
+    rm -rf /tmp/cmake
+
+# install boost 1.65.1
+RUN set -e; \
+    git clone https://github.com/boostorg/boost /tmp/boost; \
+    (cd /tmp/boost ; git checkout 436ad1dfcfc7e0246141beddd11c8a4e9c10b146); \
+    (cd /tmp/boost ; git submodule update --init --recursive); \
+    (cd /tmp/boost ; /tmp/boost/bootstrap.sh --with-libraries=system,filesystem); \
+    (cd /tmp/boost ; /tmp/boost/b2 headers); \
+    (cd /tmp/boost ; /tmp/boost/b2 cxxflags="-std=c++14" -j ${PARALLELISM} install --prefix=/opt/dependencies/boost); \
+    ldconfig; \
+    rm -rf /tmp/boost
+
+# install protobuf v3.5.1
+RUN set -e; \
+    git clone https://github.com/google/protobuf /tmp/protobuf; \
+    (cd /tmp/protobuf ; git checkout 106ffc04be1abf3ff3399f54ccf149815b287dd9); \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -Dprotobuf_BUILD_TESTS=OFF \
+        -Dprotobuf_BUILD_SHARED_LIBS=ON \
+        -H/tmp/protobuf/cmake \
+        -B/tmp/protobuf/.build \
+        -DCMAKE_INSTALL_PREFIX=/opt/dependencies/protobuf; \
+    cmake --build /tmp/protobuf/.build --target install -- -j${PARALLELISM}; \
+    ldconfig; \
+    rm -rf /tmp/protobuf
+
+# install gflags
+RUN set -e; \
+    git clone https://github.com/gflags/gflags /tmp/gflags; \
+    (cd /tmp/gflags ; git checkout f8a0efe03aa69b3336d8e228b37d4ccb17324b88); \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -H/tmp/gflags \
+        -B/tmp/gflags/build \
+        -DCMAKE_INSTALL_PREFIX=/opt/dependencies/gflags; \
+    cmake --build /tmp/gflags/build --target install -- -j${PARALLELISM}; \
+    ldconfig; \
+    rm -rf /tmp/gflags
+
+# install c-ares
+RUN set -e; \
+    git clone https://github.com/c-ares/c-ares /tmp/c-ares; \
+    (cd /tmp/c-ares ; git checkout 3be1924221e1326df520f8498d704a5c4c8d0cce); \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -H/tmp/c-ares \
+        -B/tmp/c-ares/build \
+        -DCMAKE_INSTALL_PREFIX=/opt/dependencies/c-ares; \
+    cmake --build /tmp/c-ares/build --target install -- -j${PARALLELISM}; \
+    ldconfig; \
+    rm -rf /tmp/c-ares
+
+# install grpc 1.11.0
+RUN set -e; \
+    export LD_LIBRARY_PATH=/opt/dependencies/protobuf/lib:$LD_LIBRARY_PATH; \
```

Comment:
It is better to use quotes: `"/opt/dependencies/protobuf/lib:$LD_LIBRARY_PATH"`, because previous value of `LD_LIBRARY_PATH` can contain whitespaces, which can cause problems if it is not escaped with quotes.